### DISCUSSION TITLE
jupyterlite integration

### DISCUSSION THREAD

**Discussion by wardi** [ckan/ckan#6091] [Ideas]
https://github.com/jtpio/jupyterlite runs Jupyter completely within the client's web browser. We could integrate it into a resource view that lets users immediately play with data in a safe, contained way.

We could even give access to all the data on a ckan site by creating virtual storage that presents ckan resource data as a filesystem.

https://phiresky.github.io/blog/2021/hosting-sqlite-databases-on-github-pages/  shows how data published in SQLite format can be even more powerful for visualizations or safe, scalable data exploring with SQL. If our storage back ends support partial downloads (nginx does, not sure about cloudstorage et.al) then client code can leverage sqlite's built in indexes to fetch only the parts of the db files they need.

**Comment by jqnatividad** [comment]
Perhaps, we can also create a Google Colab resource view.

By creating a CKAN-instance level Google account, we can leverage [Google Colab](https://colab.research.google.com/notebooks/intro.ipynb), and run a resource's associated notebooks on Google's cloud servers (including GPUs and TPUs) automatically as configured in the resource view (i.e. on demand, on schedule, when the resource is updated)

https://stackoverflow.com/questions/50595831/google-colab-api

Here are some experiments I did earlier last year with Google Colab and Plot.ly that show the potential of what can be done with Colab (and being able to load just about any module like Pandas and Plot.ly)

https://data.beta.nyc/pages/nyc-covid19
https://github.com/dathere/notebooks/blob/master/NYC_borough_covid19.ipynb
https://github.com/dathere/notebooks/blob/master/Realtime_R0_USA_By_County_USAFacts.ipynb
